modoboa_installer/database.py

```diff
@@ -150,6 +152,9 @@ def create_user(self, name, password):
         self._exec_query(
             "CREATE USER '{}'@'%' IDENTIFIED BY '{}'".format(
                 name, password))
+        self._exec_query(
+            "CREATE USER '{}'@'localhost' IDENTIFIED BY '{}'".format(
+                name, password))
```

Comment:
You're right, but Mariadb/mysql are provided with another user ''@'localhost' which is more specific that 'modoboa'@'%', shadowing its credentials. 
See [https://dev.mysql.com/doc/refman/5.5/en/problems-connecting.html](MySQL 5.5 Reference Manuall) 
`If you cannot figure out why you get Access denied, remove from the user table all rows that have Host values containing wildcards (rows that contain '%' or '_' characters). A very common error is to insert a new row with Host='%' and User='some_user', thinking that this enables you to specify localhost to connect from the same machine. The reason that this does not work is that the default privileges include a row with Host='localhost' and User=''. Because that row has a Host value 'localhost' that is more specific than '%', it is used in preference to the new row when connecting from localhost! The correct procedure is to insert a second row with Host='localhost' and User='some_user', or to delete the row with Host='localhost' and User=''. `